### PR TITLE
feat(frontend): Migrate Explorer to the Indexer for Explorer

### PR DESCRIPTION
**Pull request by icerove**
migration plan frontend part.
finish #450 2 3 and 4 part

### CODE REVIEW

**Comment by frol** [comment]
The tests fail with:

```
src/components/accounts/AccountRow.tsx(2,1): error TS6133: 'moment' is declared but its value is never read.
src/components/blocks/__tests__/BlockDetails.test.tsx(11,13): error TS2322: Type '{ transactionsCount: number; timestamp: number; hash: string; prevHash: string; height: number; gasPrice: string; gasUsed: string; isFinal: true; }' is not assignable to type 'BlockInfo'.
  Object literal may only specify known properties, but 'transactionsCount' does not exist in type 'BlockInfo'. Did you mean to write 'transactions_count'?
src/components/blocks/__tests__/BlocksRow.test.tsx(11,13): error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: Readonly<Props>): default', gave the following error.
    Type '{ transactionsCount: number; timestamp: number; hash: string; prevHash: string; height: number; gasPrice: string; isFinal: false; }' is not assignable to type 'BlockInfo'.
      Object literal may only specify known properties, but 'transactionsCount' does not exist in type 'BlockInfo'. Did you mean to write 'transactions_count'?
  Overload 2 of 2, '(props: Props, context?: any): default', gave the following error.
    Type '{ transactionsCount: number; timestamp: number; hash: string; prevHash: string; height: number; gasPrice: string; isFinal: false; }' is not assignable to type 'BlockInfo'.
      Object literal may only specify known properties, but 'transactionsCount' does not exist in type 'BlockInfo'. Did you mean to write 'transactions_count'?
src/components/dashboard/DashboardBlocksBlock.tsx(38,34): error TS2551: Property 'transactionsCount' does not exist on type 'BlockInfo'. Did you mean 'transactions_count'?
src/components/dashboard/__tests__/BlockBlock.test.tsx(11,13): error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: Readonly<Props>): default', gave the following error.
    Type '{ transactionsCount: number; timestamp: number; hash: string; prevHash: string; height: number; gasPrice: string; isFinal: true; }' is not assignable to type 'BlockInfo'.
      Object literal may only specify known properties, but 'transactionsCount' does not exist in type 'BlockInfo'. Did you mean to write 'transactions_count'?
  Overload 2 of 2, '(props: Props, context?: any): default', gave the following error.
    Type '{ transactionsCount: number; timestamp: number; hash: string; prevHash: string; height: number; gasPrice: string; isFinal: true; }' is not assignable to type 'BlockInfo'.
      Object literal may only specify known properties, but 'transactionsCount' does not exist in type 'BlockInfo'. Did you mean to write 'transactions_count'?
```

Please, address that

File: frontend/src/libraries/explorer-wamp/accounts.ts [hunk 1]
```diff
@@ -82,6 +89,24 @@ export default class AccountsApi extends ExplorerApi {
...
+            LIMIT :limit`,
+          {
+            limit,
+            ...paginationIndexer,
+          },
+        ]);
+      }
       return await this.call("select", [
         `SELECT account_id as accountId, created_at_block_timestamp as createdAtBlockTimestamp, 
           created_by_transaction_hash as createdByTransactionHash, account_index as accountIndex
```
Comment (frol) [inline frontend/src/libraries/explorer-wamp/accounts.ts:112]: The schema on Indexer is different from the one you use here. Please, double-check that this is even executed.

Comment (icerove) [reply frontend/src/libraries/explorer-wamp/accounts.ts:113]: I know, and it is not easy as it changed a lot other than blocks, I will think a good solution here

**Comment by icerove** [comment]
fix the query method but wait for your new design for call current database or indexer. @frol

File: frontend/src/components/transactions/ActionMessage.tsx [hunk 2]
```diff
@@ -122,25 +121,36 @@ const transactionMessageRenderers: TransactionMessageRenderers = {
...
-              actionArgs.access_key.permission.FunctionCall.method_names
-                .length > 0
-                ? `(${actionArgs.access_key.permission.FunctionCall.method_names.join(
-                    ", "
-                  )})`
-                : "any"
-            } methods and nonce ${actionArgs.access_key.nonce}`}
-          </p>
-        </>
+        actionArgs.access_key.permission.permission_kind ? (
```
Comment (frol) [inline frontend/src/components/transactions/ActionMessage.tsx:124]: Is it some sort of Indexer support? Let's try to address the distinction in the `libraries/explorer-wamp/*` scope, so the rest of the code is not polluted with the changes from the switch of the data source. Please, apply the same reasoning throughout the migration.

File: frontend/src/components/utils/ListHandler.tsx [hunk 3]
```diff
@@ -113,12 +113,17 @@ export default (
...
+            ? {
+                accountIndex: this.state.items[this.state.items.length - 1]
+                  .accountIndex,
+              }
+            : {
+                endTimestamp: this.state.items[this.state.items.length - 1]
+                  .createdAtBlockTimestamp,
+                accountIndex: this.state.items[this.state.items.length - 1]
+                  .accountIndex,
+              };
```
Comment (frol) [inline frontend/src/components/utils/ListHandler.tsx:126]: I assume that we only use `accountIndex` here, so I think we should consider just define the `accountIndex` as `{ createdAtBlockTimestamp, accountIndex }` when we use the old data source, and `{ accountIndex }` when we use the new one in the `libraries/explorer-wamp/*`, so we don't need to have `isIndexer` and this error-prone implementation here.

File: frontend/src/components/utils/Timer.jsx [hunk 4]
```diff
@@ -4,7 +4,7 @@ export default class extends React.PureComponent {
   constructor(props) {
     super(props);
 
-    const time = props.time === undefined ? new Date() : props.time;
+    let time = props.time === undefined ? new Date() : props.time;
```
Comment (frol) [inline frontend/src/components/utils/Timer.jsx:7]: Why?

File: frontend/src/components/utils/Timer.jsx [hunk 5]
```diff
@@ -20,7 +20,7 @@ export default class extends React.PureComponent {
   }
 
   componentDidUpdate() {
-    const time = this.props.time === undefined ? new Date() : this.props.time;
+    let time = this.props.time === undefined ? new Date() : this.props.time;
```
Comment (frol) [inline frontend/src/components/utils/Timer.jsx:23]: ^

**Review by frol — COMMENTED** [review]
@icerove This is massive! Thanks for this tremendous work! I will work on the new design later today

File: frontend/src/components/transactions/ActionMessage.tsx (hunk 2 above)
Comment (icerove) [reply frontend/src/components/transactions/ActionMessage.tsx:124]: the current args in database: 
`{"public_key":"ed25519:F5zJLKH4V5Lyw9CvAtc6UVh7n5YhPaFKhdn2ZyzsGivL","access_key":{"nonce":0,"permission":"FullAccess"}}`

the args in indexer: 
`{"access_key": {"nonce": 0, "permission": {"permission_kind": "FULL_ACCESS"}}, "public_key": "ed25519:Ff93hQx7HBfPrAZ4uHgFBTWezB53NyrpG5SM8C2G1V2"}`

it is not straightforward to address the distinction here

Comment (icerove) [reply frontend/src/components/transactions/ActionMessage.tsx:124]: @frol

**Review by frol — APPROVED** [review]
Blocked by #499

**Pull request merged**